exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/ElasticsearchClient.java

```diff
@@ -203,27 +203,70 @@ public boolean putIndexTemplate(
     template.put("index_patterns", Collections.singletonList(templateName + INDEX_DELIMITER + "*"));
     template.put("composed_of", Collections.singletonList(configuration.index.prefix));
 
-    // update alias in template in case it was changed in configuration
-    final Object templateProperties =
-        template.computeIfAbsent("template", key -> new HashMap<String, Object>());
-    if (templateProperties instanceof Map) {
-      final Map templatePropertyMap = (Map) templateProperties;
-      templatePropertyMap.put(
-          "aliases", Collections.singletonMap(aliasName, Collections.emptyMap()));
+    // update aliases in template in case it was changed in configuration
+    final Map<String, Object> templateProperties = getTemplatePropertyMap(template, templateName);
+    templateProperties.put("aliases", Collections.singletonMap(aliasName, Collections.emptyMap()));
+
+    // update number of shards in template in case it was changed in configuration
+    final Map<String, Object> settingsProperties =
+        getSettingsPropertyMap(templateProperties, templateName);
+    updateSettings(settingsProperties);
+
+    return putIndexTemplate(templateName, template);
+  }
+
+  private Map<String, Object> getTemplatePropertyMap(
+      Map<String, Object> properties, String templateName) {
+    return getProperties(properties, "template", templateName);
+  }
+
+  private Map<String, Object> getSettingsPropertyMap(
+      Map<String, Object> properties, String templateName) {
+    return getProperties(properties, "settings", templateName);
+  }
+
+  private Map<String, Object> getProperties(
+      Map<String, Object> from, String fieldName, String templateName) {
+    final Object properties = from.computeIfAbsent(fieldName, key -> new HashMap<String, Object>());
```

Comment:
:question: Why do you use `computeIfAbsent()`? What about `getOrDefault()`? Is there a reason to write the empty map back into from?

Reply:
Oh, I think I get it now. I think you do it to add a map on the settings field so that when you update the settings, then magically the settings are part of the tree of maps.

:x: Please change that. 
* In my mind, the `getProperties()` method should not modify the map. This is very unexpected behavior
* And either the `updateSettings` method should modify the settings in the template, or `putIndexTemplate` should write the settings map back into the template.

One alternative would be to rename the method to `getOrCreateProperties()`

Reply:
Thanks for the hint, I renamed the methods to `getOrCreate...()`.